hist/hist/src/TF2.cxx

```diff
@@ -842,35 +842,40 @@ void TF2::Save(Double_t xmin, Double_t xmax, Double_t ymin, Double_t ymax, Doubl
 void TF2::SavePrimitive(std::ostream &out, Option_t *option /*= ""*/)
 {
    char quote = '"';
+   TString f2Name(GetName());
    out<<"   "<<std::endl;
    if (gROOT->ClassSaved(TF2::Class())) {
       out<<"   ";
    } else {
       out<<"   TF2 *";
    }
    if (!fMethodCall) {
-      out<<GetName()<<" = new TF2("<<quote<<GetName()<<quote<<","<<quote<<GetTitle()<<quote<<","<<fXmin<<","<<fXmax<<","<<fYmin<<","<<fYmax<<");"<<std::endl;
+      out<<f2Name.Data()<<" = new TF2("<<quote<<f2Name.Data()<<quote<<","<<quote<<GetTitle()<<quote<<","<<fXmin<<","<<fXmax<<","<<fYmin<<","<<fYmax<<");"<<std::endl;
    } else {
-      out<<GetName()<<" = new TF2("<<quote<<GetName()<<quote<<","<<GetTitle()<<","<<fXmin<<","<<fXmax<<","<<fYmin<<","<<fYmax<<","<<GetNpar()<<");"<<std::endl;
+      out<<f2Name.Data()<<" = new TF2("<<quote<<f2Name.Data()<<quote<<","<<GetTitle()<<","<<fXmin<<","<<fXmax<<","<<fYmin<<","<<fYmax<<","<<GetNpar()<<");"<<std::endl;
    }
 
-   SaveFillAttributes(out, GetName(), 0, 1001);
-   SaveMarkerAttributes(out, GetName(), 1, 1, 1);
-   SaveLineAttributes(out, GetName(), 1, 1, 4);
+   SaveFillAttributes(out, f2Name.Data(), 0, 1001);
+   SaveMarkerAttributes(out, f2Name.Data(), 1, 1, 1);
+   SaveLineAttributes(out, f2Name.Data(), 1, 1, 4);
 
    if (GetNpx() != 100)
-      out<<"   "<<GetName()<<"->SetNpx("<<GetNpx()<<");"<<std::endl;
+      out<<"   "<<f2Name.Data()<<"->SetNpx("<<GetNpx()<<");"<<std::endl;
```

Comment:
Why only custom `npx` is preserved? I guess, `npy` is missing here.

Reply:
I'll look .. I do not change that code... I was like that.